torch/_export/graph_module.py

```diff
@@ -224,61 +193,131 @@ def op_to_str(op):
-        input_shape_constraints=input_shape_constraints_by_src_name,
-        inline_constraints=inline_constraints,
-        input_name_to_example_inputs=input_name_to_example_inputs,
-    )
-    attach_export_graph_metadata(gm, meta)
+) -> torch.fx.GraphModule:
+    gm = torch.fx.GraphModule(root, graph, class_name=class_name)
+    gm.__class__ = type(type(gm).__name__, (ExportGraphModuleMixin, type(gm)), {})
     return gm
 
 
+class ExportedProgram:
+    def __init__(
+        self,
+        root: Union[torch.nn.Module, Dict[str, Any]],
+        graph: torch.fx.Graph,
+        graph_signature: Optional[GraphSignature] = None,
+        call_spec: Optional[CallSpec] = None,
+        state_dict: Optional[Dict[str, Any]] = None,
+        example_inputs: Any = None,
+    ):
+        # Remove codegen related things from the graph. It should just be a flat graph.
+        graph._codegen = torch.fx.graph.CodeGen()
+        graph_module = make_export_graph_module(root, graph)
+        self.graph_module: ExportGraphModule = graph_module
+
+        self.graph_signature: Optional[GraphSignature] = graph_signature
+        self.call_spec: Optional[CallSpec] = call_spec
+        self.state_dict: Optional[Dict[str, Any]] = state_dict
+        self.symbol_to_range = {}
+
+        input_shape_constraints = (
+            root.meta.get("input_shape_constraints", [])
+            if isinstance(root, torch.fx.GraphModule)
+            else []
+        )
+        self.symbol_to_range = (
+            root.meta.get("inline_constraints", {})
+            if isinstance(root, torch.fx.GraphModule)
+            else {}
+        )
+
+        # group by input id
+        input_shape_constraints_by_tensor_id = defaultdict(list)
+        for constraint in input_shape_constraints:
+            input_shape_constraints_by_tensor_id[constraint["t_id"]].append(
+                (constraint["dim"], constraint["min"], constraint["max"])
+            )
+
+        input_shape_constraints_by_src_name: Dict[str, List[Tuple[int, ConstraintExpr, ConstraintExpr]]] = {}
+        input_name_to_example_inputs: Dict[str, Any] = {}
+        if example_inputs is not None:
+            input_tracker = 0
+            for node in graph_module.graph.nodes:
+                if node.op == "placeholder":
+                    example_input = example_inputs[input_tracker]
+                    if id(example_input) in input_shape_constraints_by_tensor_id:
+                        input_shape_constraints_by_src_name[node.name] = input_shape_constraints_by_tensor_id[id(example_input)]
+                    input_name_to_example_inputs[node.name] = example_input
+                    input_tracker += 1
+
+        self._input_shape_constraints = input_shape_constraints_by_src_name
+        self._input_name_to_example_inputs = input_name_to_example_inputs
+
+    def __call__(self, *args: Any):
+        return self.forward(*args)
+
+    def forward(self, *args: Any) -> Any:
```

Comment:
Is there any reason why want to abstract away this call? Why not just inline in __call__?